### PR TITLE
feat(opensearch serverless): analyzer

### LINKED ISSUE #536: (opensearch serverless): (adding analyzer)
### Describe the feature

Adding [analyzers](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/serverless-genref.html). For the case in Japanese use, default analyzer is totally useless. One of the motivation to use OpenSearch is run HYBRID search, so this feature is critical for Japanese users.

### Use Case

Hybrid search for Japanese documents

### Proposed Solution

_No response_

### Other Information

_No response_

### Acknowledgements

- [x] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### PR DESCRIPTION
**Pull request by statefb**
close #536 

Add analyzers support for the construct: https://opensearch.org/docs/latest/analyzers/

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the project license.

### CODE REVIEW

**Comment by statefb** [comment]
example usage:
```ts
import { bedrock } from "@cdklabs/generative-ai-cdk-constructs";
import {
  CharacterFilterType,
  TokenFilterType,
  TokenizerType,
} from "@cdklabs/generative-ai-cdk-constructs/lib/cdk-lib/opensearchserverless";

const kb = new bedrock.KnowledgeBase(this, "KB", {
    embeddingsModel: bedrock.BedrockFoundationModel.TITAN_EMBED_TEXT_V1,
    instruction:
      "Use this knowledge base to answer questions. Please quote the reference to explain your answers.",
    analyzer: {
      characterFilters: [CharacterFilterType.ICU_NORMALIZER],
      tokenizer: TokenizerType.KUROMOJI_TOKENIZER,
      tokenFilters: [
        TokenFilterType.KUROMOJI_BASEFORM,
        TokenFilterType.JA_STOP,
      ],
    }
)
```

File: src/cdk-lib/bedrock/knowledge-base.ts
Comment (krokoko) [inline src/cdk-lib/bedrock/knowledge-base.ts:1]: I don't think we need to modify the knowledge base file. By default, the knowledge base creates the vector index on the user's behalf if is not provided, so default implementation without analyzer is fine. If the user wants to customize the index, he will instantiate his own index and pass it to through ```vectorIndex``` when instantiating the Knowledge Base

Comment (statefb) [reply src/cdk-lib/bedrock/knowledge-base.ts:1]: >his own index and pass it to through

Agreed, but I'd like to implement another solution instead of that. Would you mind if I add `VectorIndex.fromAnalyzer` method?  VectorIndex looks a little bit complicated because it requires very specific knowledge about KB, which should be avoided in context of L2 construct. For example, users need to specify MappingField correctly and need to know the detail specification of KB to implement it.


```ts
new VectorIndex(this, 'KBIndex', {
          collection: this.vectorStore as VectorCollection,
          indexName,
          vectorField,
          vectorDimensions: embeddingsModel.vectorDimensions!,
          mappings: [
            {
              mappingField: 'AMAZON_BEDROCK_TEXT_CHUNK',
              dataType: 'text',
              filterable: true,
            },
            {
              mappingField: 'AMAZON_BEDROCK_METADATA',
              dataType: 'text',
              filterable: false,
            },
          ],
          analyzer: props.analyzer,
        });
```

Comment (statefb) [reply src/cdk-lib/bedrock/knowledge-base.ts:1]: @krokoko Removed Analyzer from KB construct and added [fromVectorIndexAttributes](https://github.com/awslabs/generative-ai-cdk-constructs/pull/537/files#diff-394baf2357ad315e4f3c94ad6f756cb0da9aa367b3f919c927bbe6420cf73f84R293-R325) method so that user can create the index from related attributes (The main motivation is not to force users to know the `mappings` like `AMAZON_BEDROCK_TEXT_CHUNK `)

Example usage:
```ts
const vectorCollection = new VectorCollection(this, "KBVectors");
  const vectorIndex = VectorIndex.fromVectorIndexAttributes(this, "KBIndex", {
    collection: vectorCollection,
    embeddingsModel: bedrock.BedrockFoundationModel.TITAN_EMBED_TEXT_V1,
    analyzer: {
      characterFilters: [CharacterFilterType.ICU_NORMALIZER],
      tokenizer: TokenizerType.KUROMOJI_TOKENIZER,
      tokenFilters: [
        TokenFilterType.KUROMOJI_BASEFORM,
        TokenFilterType.JA_STOP,
      ],
    },
  });
  const kb = new bedrock.KnowledgeBase(this, "KB", {
    embeddingsModel: bedrock.BedrockFoundationModel.TITAN_EMBED_TEXT_V1,
    vectorStore: vectorCollection,
    vectorIndex: vectorIndex,
    instruction:
      "Use this knowledge base to answer questions. Please quote the reference to explain your answers.",
  });

```

Comment (krokoko) [reply src/cdk-lib/bedrock/knowledge-base.ts:1]: Now we add a dependency on bedrock models in oss which I would prefer to avoid (because of ```embeddingsModel```). To me the method ```fromVectorIndexAttributes``` is confusing, as it seems to be **constructor minus mappings** field, but still specific to a knowledge base. I think it is ok to keep the constructor only to add the analyzer. We have a code snippet in the README with the mapping example for a knowledge base, so users can copy paste that maybe ? https://github.com/awslabs/generative-ai-cdk-constructs/blob/main/src/cdk-lib/opensearch-vectorindex/README.md#vector-index

**Comment by krokoko** [comment]
Also, could you please update the code snippet here: https://github.com/awslabs/generative-ai-cdk-constructs/blob/main/src/cdk-lib/opensearch-vectorindex/README.md#vector-index with the new analyzer props

**Comment by krokoko** [comment]
Build is not passing: https://github.com/awslabs/generative-ai-cdk-constructs/actions/runs/9752808672/job/26916988890?pr=537

File: src/cdk-lib/bedrock/knowledge-base.ts
Comment (statefb) [reply src/cdk-lib/bedrock/knowledge-base.ts:1]: Got it. Omitted `fromVectorIndexAttributes` and added explanation on the `README.md`.

**Review by krokoko — APPROVED** [review]

**Comment by krokoko** [comment]
> Build is not passing: https://github.com/awslabs/generative-ai-cdk-constructs/actions/runs/9752808672/job/26916988890?pr=537

fixed

**Comment by krokoko** [comment]
Great thank you for your contribution @statefb ! I approved, once a second reviewer will approve we will merge

**Review by dineshSajwan — APPROVED** [review]
LGTM

**Pull request merged**
